ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -2026,6 +2038,8 @@ protected RelNode applyMaterializedViewRewriting(RelOptPlanner planner, RelNode
 
       // Optimize plan
       if (!ruleExclusionRegex.isEmpty()) {
+        LOG.debug("The CBO rules matching the following regex are excluded from planning: {}",
```

Comment:
I would maybe argue to make these info since they could modify planning in strange ways and most users do not run with debug logging on. Sure they could run EXPLAIN on the plan to see, but sometimes the person doing the initial debugging doesn't have access to the EXPLAIN output (they just have a set of logs and a description of "queries are returning wrong results").

Anyways - this is just a suggestion and if you feel different go ahead and ignore because I can see the other argument of not wanting to add more INFO log lines.

Thanks for this work @asolimando

Reply:
I think this is a very valid point, and this property should not be on apart from very edge cases, so I don't think it will clog the output much. Let me move from debug to info as you suggest!